substrate/frame/staking/stake-tracker/src/lib.rs

```diff
@@ -0,0 +1,523 @@
+				voter_weight.into(),
+			);
+
+			// updates vote weight of nominated targets accordingly. Note: this will
+			// update the score of up to `T::MaxNominations` validators.
+			for target in nominations.into_iter() {
+				Self::update_target_score(&target, stake_imbalance);
+			}
+		}
+
+		/// Updates the stake of a target.
+		pub(crate) fn do_stake_update_target(
+			who: &T::AccountId,
+			prev_stake: Option<Stake<BalanceOf<T>>>,
+			stake: Stake<BalanceOf<T>>,
+		) {
+			let voter_weight = Self::to_vote(stake.active).into();
+			let stake_imbalance = StakeImbalance::from(
+				prev_stake.map_or(Default::default(), |s| Self::to_vote(s.active).into()),
+				voter_weight,
+			);
+
+			Self::update_target_score(who, stake_imbalance);
+
+			// validator is both a target and a voter. update the voter score if the voter list
+			// is in strict mode.
+			if T::VoterUpdateMode::get().is_strict_mode() {
+				let _ = T::VoterList::on_update(who, Self::to_vote(stake.active)).defensive_proof(
+					"the staker should exist in VoterList, as per the \
+                            contract with staking.",
+				);
+			}
+		}
+
+		/// Updates a target's score by increasing/decreasing an imbalance of the current score in
+		/// the target list.
+		pub(crate) fn update_target_score(
+			who: &T::AccountId,
+			imbalance: StakeImbalance<ExtendedBalance>,
+		) {
+			// if target list does not contain target, add it and proceed.
+			if !T::TargetList::contains(who) {
+				let _ = T::TargetList::on_insert(who.clone(), Zero::zero()).defensive_proof(
+					"staker does not yet exist in the list as per check above; qed.",
+				);
+			}
+
+			// update target score.
+			match imbalance {
+				StakeImbalance::Positive(imbalance) => {
+					let _ = T::TargetList::on_increase(who, imbalance).defensive_proof(
+						"staker should exist in the list, otherwise returned earlier.",
+					);
+				},
+				StakeImbalance::Negative(imbalance) => {
+					if let Ok(current_score) = T::TargetList::get_score(who) {
+						let balance = current_score.saturating_sub(imbalance);
+
+						// the target is removed from the list IFF score is 0.
+						if balance.is_zero() {
+							let _ = T::TargetList::on_remove(who).defensive_proof(
+								"staker exists in the list as per the check above; qed.",
+							);
+						} else {
+							// update the target score without removing it.
+							let _ = T::TargetList::on_update(who, balance).defensive_proof(
+								"staker exists in the list as per the check above; qed.",
+							);
+						}
+					} else {
+						defensive!("unexpected: unable to fetch score from staking interface of an existent staker");
+					}
+				},
+			};
+		}
+
+		// ------ Helpers
+
+		/// Helper to convert the balance of a staker into its vote weight.
+		pub(crate) fn to_vote(balance: BalanceOf<T>) -> VoteWeight {
+			<T::Staking as StakingInterface>::CurrencyToVote::to_vote(
+				balance,
+				T::Currency::total_issuance(),
+			)
```

Comment:
`total_issuance` is not constant, can this lead to error? like the inserted vote weight is different from the vote weight calculated with the new total issuance and so the list is unordered in regards to actual staked amount?